packages/react/src/modules/slack/hooks/useSlackChannels.ts

```diff
@@ -0,0 +1,80 @@
+import Knock, { User } from "@knocklabs/client";
+import { useState, useEffect, useMemo } from "react";
+
+export interface Channel {
+  name: string;
+  id: string;
+  connected: boolean;
+}
+
+export type ContainerObject = {
+  objectId: string;
+  collection: string;
+};
+
+export function useSlackChannels(
+  tenant: string,
+  connectionsObject: ContainerObject,
+  host: string,
+  knockClientId: string,
+  knockSlackChannelId: string,
+  userToken: string,
+  user: User,
+  shouldRefetch: boolean,
+  setHasError: (hasError: boolean) => void,
+  hasError: boolean,
+): {
+  data: { channels: Channel[] };
+  isLoading: boolean;
+  error: boolean;
+} {
+  const [isLoading, setIsLoading] = useState(true);
+  const [data, setData] = useState<{ channels: Channel[] }>({
+    channels: [],
+  });
+
+  const knockClient = useMemo(() => {
```

Comment:
Same as the picker component, I think we might be able to rely on `useKnock` to get the client.